React/Views/RCTViewManager.m

```diff
@@ -255,7 +255,15 @@ - (RCTViewManagerUIBlock)uiBlockToAmendWithShadowViewRegistry:(__unused NSDictio
 RCT_VIEW_BORDER_RADIUS_PROPERTY(BottomLeft)
 RCT_VIEW_BORDER_RADIUS_PROPERTY(BottomRight)
 
-RCT_REMAP_VIEW_PROPERTY(zIndex, reactZIndex, NSInteger)
+RCT_CUSTOM_VIEW_PROPERTY(zIndex, NSInteger, RCTView)
```

Comment:
I believe we have to move this logic to `RCTView` and `UIView+React` introducing new category method `reactZIndex`. (Where we also have to reset the cache for superview... So we also have to have something like `reactInvalidateZIndex`.)